buildScript/webpack.config.js

```diff
@@ -130,102 +221,47 @@ var webpackConfig = {
-    //
-    //        loader.loader = ExtractTextPlugin.extract(first, rest.join('!'));
-    //        delete loader.loaders;
-    //    }
-    //
-    //    return loader;
-    //});
 
-    webpackConfig.plugins.push(
-        new webpack.optimize.UglifyJsPlugin({
-            compress : {
-                warnings  : false,
-                unused    : true,
-                dead_code : true
-            }
-        })
-    );
-}
 
-var buildCnt=0;
+export default makeWebpackConfig;
 
-if (globals.__DEBUG__) {
-    const progressDone= function() {
-        buildCnt++;
-        process.stdout.write('\n');
-        var time = new Date();
-        var tStr= time.getHours() + ':' + time.getMinutes() + ':' + time.getSeconds();
-        process.stdout.write('-------------------- Begin build #'+ buildCnt +
-                             ' at '+ tStr +'--------------------------\n');
-        //process.stdout.write('Build ' +buildCnt+ ' results: '+ tStr);
-    };
 
-    webpackConfig.plugins.splice(3,0,
-        new ProgressBarPlugin({
-            callback : progressDone
-        })
-    );
-}
 
-// ------------------------------------
-// Optional Configuration
-// ------------------------------------
-
-// if index_html exists, insert script tag to load built javascript bundles(s).
-if (fs.existsSync(path.resolve(config.dist, config.index_html))) {
-    webpackConfig.plugins.push(
-        new HtmlWebpackPlugin({
-            template : path.resolve(config.src, config.html_dir, config.index_html),
-            hash     : false,
-            filename : config.index_html,
-            minify   : globals.__PROD__,
-            inject   : 'body'
-        })
-    );
-}
+function firefly_loader(loadScript, outpath, debug=true) {
+    return function () {
+        this.plugin('done', function (stats) {
+            // console.log(Object.keys(stats.compilation));
+            var hash = debug ? 'dev' : stats.hash;
 
-if (config.do_lint) {
-    let eslint_options = '';
-    if (process.env.DO_LINT_STRICT) {
-        // in addition to .eslintrc, extra rules are defined in .eslint-strict.json
-        const eslint_strict_path = path.resolve(config.project, '.eslint-strict.json');
-        if (fs.existsSync(eslint_strict_path)) {
-            eslint_options = '?' + JSON.stringify(JSON.parse(fs.readFileSync(eslint_strict_path)));
-            console.log('eslint-loader' + eslint_options);
-        } else {
-            console.log('ERROR: No .eslint-strict.json found - excluding lint');
-            console.log('----------------------------------------------------');
-            config.do_lint = false;
-        }
-    }
-    if (config.do_lint) {
-    	webpackConfig.module.preLoaders = [
-        	{
-            	test : /\.(js|jsx)$/,
-            	exclude: exclude_dirs,
-                loaders: ['eslint-loader' + eslint_options]
-        	}
-    	];
-	}
-}
+            console.log('!!!!!!!!' + path.resolve(outpath, `${stats.compilation.name}.nocache.js`));
```

Comment:
Should it be here? Build script outputs
...
!!!!!!!!/hydra/cm/firefly/build/firefly/gwt/firefly/firefly.nocache.js
good!!!!!!
after good!!!!!!
...
Looks like leftovers from debugging.